pkg/utils/config.go

```diff
@@ -0,0 +1,133 @@
+package utils
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	color "github.com/threeal/bro/pkg/utils/color"
+)
+
+type Config interface {
+	Read() error
+	Write() error
+	init()
+}
+
+var HOME_DIR = getHomeDirectory()
+var BRO_CONFIG_DIR = filepath.Join(*HOME_DIR, ".bro")
+
+type BackendConfig struct {
+	ListenAddr *string `json:"listen_addr"`
+}
+
+type ClientConfig struct {
+	BackendAddr *string `json:"backend_addr"`
+}
+
+func getHomeDirectory() *string {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		log.Fatalf("failed to determine home directory: %v", err)
+	}
+	return &homeDir
+}
```

Comment:
We should always pass an error in function to the caller, and let the caller do something if this function is failed.
E.g. like in this case, we don't solve anything if `os.UserHomeDir()` is failed, the return value is still wrong, and we let the program to be running. It's better to make it crash rather than letting it to run with a wrong behavior.

Reply:
log.Fatalf() calls exit() iirc

Reply:
so the program will simply crash and the function will not return wrong value @threeal

Reply:
ooh right, sorry for not noticing that lol.

But still, it's better to just return error, since we need to test this function, and we couldn't test if this function just exit immediately.